pkg/util/prometheus/prometheus_struct.go

```diff
@@ -3,35 +3,60 @@ package prometheus
 import (
 	"encoding/json"
 	"fmt"
+	"net/url"
 	"strings"
 
 	"github.com/maistra/maistra-test-tool/pkg/util/oc"
 	"github.com/maistra/maistra-test-tool/pkg/util/pod"
 	"github.com/maistra/maistra-test-tool/pkg/util/test"
 )
 
+func NewPrometheus(selector, containerName string) Prometheus {
+	return &prometheus_struct{selector, containerName}
+}
+
 type prometheus_struct struct {
-	selector string
+	selector      string
+	containerName string
+}
+
+func (pi *prometheus_struct) clone() *prometheus_struct {
+	new := *pi
+	return &new
 }
 
 var _ Prometheus = &prometheus_struct{}
 
-func NewPrometheus(selector string) Prometheus {
-	return &prometheus_struct{selector}
+func (pi *prometheus_struct) WithSelector(selector string) Prometheus {
+	new := pi.clone()
+	new.selector = selector
+	return new
+}
+
+func (pi *prometheus_struct) WithContainerName(containerName string) Prometheus {
+	new := pi.clone()
+	new.containerName = containerName
+	return new
 }
 
 func (pi *prometheus_struct) Query(t test.TestHelper, ns string, query string) PrometheusResponse {
-	escapedQuery := strings.ReplaceAll(query, `'`, `'\\''`)
+	queryString := url.Values{"query": []string{query}}.Encode()
+	url := fmt.Sprintf(`http://localhost:9090/api/v1/query?%s`, queryString)
+	url_shellEscaped := strings.ReplaceAll(url, `'`, `'\\''`)
```

Comment:
nit: In Go we use camel case.
```suggestion
	urlShellEscaped := strings.ReplaceAll(url, `'`, `'\\''`)
```